### ISSUE TITLE
Can't set directory with roms

### ISSUE THREAD

**Issue by Sunderland93** [flathub/org.duckstation.DuckStation#8]
I can't set directory with roms, no matter where it is located - on /home or any other. Run Duckstation Qt, go to Settings -> Add game directory. Choose any of them and click "Open". Nothing is happen.

**Comment by Sunderland93** [comment]
@j8r yes, it doesn't work. I think the issue is in xdg-desktop portal https://github.com/flatpak/xdg-desktop-portal/issues/594 https://github.com/flatpak/xdg-desktop-portal/issues/591

**Comment by manero6** [comment]
This happens because the application by default has no particular Filesystem permission.
This means that it cannot access files on the home directory for example.

This can be changed with a tool like [Flatseal](https://github.com/tchx84/Flatseal).

In my case my games are in `~/Games`.
I therefore used Flatseal to grant Duckstation permission to access that folder.
As I just need Duckstation to read the content, I set the path to read only this way: `~/Games:ro`.

Step-by-step:
- install Flatseal
- launch Flatseal and select the app to modify
- scroll to the "Filesystem" section
- add the desired path(s) under "Other Files"
  - for example `~/Documents` or `~/Downloads:ro`
- to grant complete home access simply activate the "All user files" switch

**Comment by j8r** [comment]
The application has `--filesystem=host:ro` by default.
It might be an issue with XDG portals then.

**Comment by manero6** [comment]
True, I just removed my custom Filesystem permission and Duckstation is able to load all games.

Another issue might be the way that paths are added through the GUI.
In my case when I add a game directory, it results in a path pointing to "/run/user/...".

If I add `~/Games/psx` with the GUI for example, I get this kind of paths:

- `/run/user/1000/doc/<hash>/psx`

While this work, the `<hash>` part of the path might change when the system restarts, and Duckstation cannot find this path anymore. One could re-add the new path, but this way there are going to be issues with the memory card.

![Screenshot from 2023-05-23 22-34-00](https://github.com/flathub/org.duckstation.DuckStation/assets/21221315/9ef49cb4-9cf4-4ff8-9eb3-2b147c35aa23)

I am able to add the correct path by modifying `~/.var/app/org.duckstation.DuckStation/data/duckstation/settings.ini` manually. Everything then works as expected.

I am not sure that every distro suffer this issue, this happens for me on Fedora 38.

ps. adding `home` through the GUI adds the correct path (that is `/home/<username>`)
